src/MizarData/include/MizarData/MizarPairedData.h

```diff
@@ -64,37 +69,33 @@ class MizarPairedDataTmpl {
   [[nodiscard]] std::vector<RelativeTimeNs> ActiveInvocationTimes(
       const absl::flat_hash_set<TID>& tids, FrameTrackId frame_track_id,
       RelativeTimeNs min_relative_time, RelativeTimeNs max_relative_time) const {
-    const auto [min_timestamp_ns, max_timestamp_ns] =
-        RelativeToAbsoluteTimestampRange(min_relative_time, max_relative_time);
-    const std::vector<TimestampNs> frame_starts =
-        GetFrameStarts(frame_track_id, min_timestamp_ns, max_timestamp_ns);
-    if (frame_starts.size() < 2) return {};
-
-    const RelativeTimeNs sampling_period = data_->GetNominalSamplingPeriodNs();
-
-    std::vector<RelativeTimeNs> result;
-    for (size_t i = 0; i + 1 < frame_starts.size(); ++i) {
-      const uint64_t callstack_count = std::transform_reduce(
-          std::begin(tids), std::end(tids), uint64_t{0}, std::plus<>(),
-          [this, i, &frame_starts](const TID tid) {
-            return CallstackSamplesCount(tid, frame_starts[i], frame_starts[i + 1]);
-          });
-
-      result.push_back(Times(sampling_period, callstack_count));
-    }
-    return result;
+    return ReduceOverFrames<std::vector<RelativeTimeNs>>(
```

Comment:
I think this reads fine, while I would have probably introduced a ForEachFrame, and perform the operation in a more iterative fashion.

Reply:
`ForEachFrame` is too general. It communicated no intent. Every call-site we currently have would have to keep track of accumulator lifetime and its updates on its own. Hence, I expose Reduce.

Reply:
As said, that would be the "iterative"-ish (you still have a consume function) equivalent. And as said. I am fine with both.